### PR TITLE
Np 45101 add verification status to cristin contributor

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 1]
```diff
@@ -104,10 +105,17 @@ public static CristinContributor randomContributor(Integer contributorIndex) {
                    .withIdentifier(contributorIndex)
                    .withGivenName(randomString())
                    .withFamilyName(randomString())
+                   .withVerificationStatus(randomVerificationStatus())
                    .withAffiliations(randomAffiliations())
                    .build();
     }
 
+    private static VerificationStatus randomVerificationStatus() {
+        return (new Random().nextInt() < 0)
```
Comment (anetteOlli) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:114]: randomBoolean?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java [hunk 2]
```diff
@@ -26,13 +28,9 @@
...
-@Builder(
-    builderClassName = "CristinContributorBuilder",
-    toBuilder = true,
-    builderMethodName = "builder",
-    buildMethodName = "build",
-    setterPrefix = "with"
-)
+@Builder(builderClassName = "CristinContributorBuilder", toBuilder = true, builderMethodName = "builder",
+    buildMethodName = "build", setterPrefix = "with")
+@JsonIgnoreProperties({"identified_cristin_person"})
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:33]: You don't need the JsonIgnoreProperty now that you provide a class for it.

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import static no.unit.nva.cristin.lambda.constants.MappingConstants.NVA_API_DOMAIN;
 import static no.unit.nva.cristin.lambda.constants.MappingConstants.PERSON_PATH;
 import static no.unit.nva.cristin.lambda.constants.MappingConstants.SHOULD_CREATE_CONTRIBUTOR_ID;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:8]: Is the actual mapping to a NVA contributor coming in a different PR?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java (hunk 2 above)
Comment (truhacevkir) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:33]: Without this @JsonIgnoreProperty I am getting error message that says unknown not ignored field.

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java (hunk 3 above)
Comment (truhacevkir) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:8]: There is already mapping from CristinContributor to Contributor, I have only added this method to extract verificationStatus                                 `.withVerificationStatus(extractVerificationStatus())`

```
    public Contributor toNvaContributor() {
        String fullName = constructFullName();
        Identity identity = new Identity.Builder()
                                .withName(fullName)
                                .withId(constructId())
                                .withVerificationStatus(extractVerificationStatus())
                                .build();

        return new Contributor.Builder()
                   .withIdentity(identity)
                   .withCorrespondingAuthor(false)
                   .withAffiliations(extractAffiliations())
                   .withRole(extractRoles())
                   .withSequence(contributorOrder)
                   .build();
    }
```

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:33]: Then your constructor is not taking it in properly.

Comment (truhacevkir) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:33]: Daniel have added two new values, the one we map is "verified_person", the one we ignore is "identified_cristin_person".

Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributor.java:33]: Ah, ok.

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**